src/coreservices.cpp

```diff
@@ -466,10 +483,17 @@ void CoreServices::initialize(QApplication* pApp) {
 
     // Load tracks in args.qlMusicFiles (command line arguments) into player
     // 1 and 2:
+
     const QList<QString>& musicFiles = m_cmdlineArgs.getMusicFiles();
+
     for (int i = 0; i < (int)m_pPlayerManager->numDecks() && i < musicFiles.count(); ++i) {
         if (SoundSourceProxy::isFileNameSupported(musicFiles.at(i))) {
             m_pPlayerManager->slotLoadToDeck(musicFiles.at(i), i + 1);
+        } else if (isPlaylist(musicFiles.at(i))) {
+            qWarning() << "Detected Playlist file!!";
+            const QString playlistFile = musicFiles.at(i);
+            qWarning() << playlistFile;
+            BasePlaylistFeature::slotCreateImportPlaylistDirect(playlistFile);
```

Comment:
You need here an instance of BasePlaylistFeature. The slot prefix is used for Qt's signal slot connections, not here. 

I think you may create something like: 
```
void Library::importPlaylistFile(..) {
    m_pPlaylistFeature->importPlaylistFile(..);
}
```

and then you can call here: 
```
m_pLibrary->importPlaylistFile(playlistFile);
```

Reply:
Or just store the playlist fie path in `CmdlineArgs` and let `AutoDJFeature` constructor deal with it?

Reply:
> You need here an instance of BasePlaylistFeature. The slot prefix is used for Qt's signal slot connections, not here.
> 
> I think you may create something like:
> 
> ```
> void Library::importPlaylistFile(..) {
>     m_pPlaylistFeature->importPlaylistFile(..);
> }
> ```
> 
> and then you can call here:
> 
> ```
> m_pLibrary->importPlaylistFile(playlistFile);
> ```



> You need here an instance of BasePlaylistFeature. The slot prefix is used for Qt's signal slot connections, not here.
> 
> I think you may create something like:
> 
> ```
> void Library::importPlaylistFile(..) {
>     m_pPlaylistFeature->importPlaylistFile(..);
> }
> ```
> 
> and then you can call here:
> 
> ```
> m_pLibrary->importPlaylistFile(playlistFile);
> ```

in the above statement `m_pPlaylistFeature->importPlaylistFile(..);` gives
`pointer to incomplete class type "PlaylistFeature" is not allowed`.
Is there a work-around here @daschuer ?

Reply:
Tbh, you could figure this out on your own by searching the web for "pointer to incomplete class type not allowed".
You need to include the headers of classes you intend to use.

Reply:
Sorry, I got confused in the cyclic dependency of classes. I'll keep that in mind, for future instances. Thanks!!